test/tensor.cpp

```diff
@@ -101,12 +101,67 @@ void test_read_write(const vector<T>& x)
     ASSERT_EQ(result1, result2);
 }
 
+template <typename T>
+void test_copy_to_same_backend(const vector<T>& x)
```

Comment:
You should pass in shape as well so you can verify that test fails if shapes don't match.